libifupdown/lifecycle.c

```diff
@@ -237,8 +240,17 @@ lif_lifecycle_run_phase(const struct lif_execute_opts *opts, struct lif_interfac
 	if (!handle_commands_for_phase(opts, envp, iface, phase))
 		goto handle_error;
 
+	/* Check if scripts dir for this phase is present and bail out if it isn't */
+	struct stat dir_stat;
+	char dir_path[4096];
+	snprintf (dir_path, 4096, "/etc/network/if-%s.d", phase);
+
+	if (stat (dir_path, &dir_stat) != 0 || S_ISDIR (dir_stat.st_mode) == 0) {
+		goto handle_error;
```

Comment:
we want to return `true` here because otherwise it will be treated as if there were a fatal error.